### ISSUE TITLE
twisted.names.cache.CacheResolver never times out data it is initialized with

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5638]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5638|
|Type|defect|
|Created|2012-04-11 12:59:14Z|
|Branch|https://github.com/twisted/twisted/tree/cacheresolver-init-5638|

The CacheResolver internal cache can be initialized with beginning values. Unlike cache entries added later, these initial entries will never be timed out or removed.

Attachments:

* [expire-init.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/31e/31e1d6413e4cc0ea30937ddb9bb5b31eff99bd4c/6bfea7f650a251dacd5d535b2e158ce07a1e3cb5.diff) (3553 bytes) - added by acidfoo on 2012-04-17 08:36:24Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5638 5638
type__defect defect
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_cacheresolver_init_5638 branches/cacheresolver-init-5638
branch_author__itamarst__acidfoo itamarst, acidfoo
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1334149154000000 1334149154000000
changetime__1336578100000000 1336578100000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="acidfoo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|acidfoo commented|
|-|-|

Changed the constructor to pass the initial entries through cacheResult instead of just assigning the dictionary. 

The original unit test (test_lookup) passes in an empty record ([], [], []) which would originally crash the cacheResult function. I wasn't sure if I could just remove the old test, so I made it assume a TTL of 0 when there are no entries in the payload.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @itamarst|
|-|-|

Thanks for the patch, looks good, with two minor issues:

  1. Extra whitespace at end of lines.
  1. A comment explaining why time elapsed of 40 is good enough to expire the record in the new test would be helpful.

I didn't realise the interface involves passing in absolute timestamps, which is pretty terrible. So I also suggest deprecating passing in cache to the constructor. That can be done in a separate ticket, though. So unless I hear from you this weekend with new patch, I will just commit this one doing the above minor tweaks myself, and open a new ticket for the deprecation.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

(In [34351]) Branching to 'cacheresolver-init-5638'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [34359]) Merge cacheresolver-init-5638: Entries added to CacheResolver's __init__ are now timed out.

Author: acidfoo
Review: itamar
Fixes: [#5638](https://github.com/twisted/twisted/issues/5638)

Fix issue where entries added via CacheResolver's constructor were never timed out.
